backend/ml/tests/test_mehestan_individual.py

```diff
@@ -0,0 +1,138 @@
+import random
+
+import pandas as pd
+import pytest
+
+from ml.mehestan.individual import compute_individual_score
+from ml.mehestan.individual_bbt import compute_individual_score as compute_individual_score_bbt
+
+
+@pytest.mark.parametrize(
+    "compute_method", [compute_individual_score, compute_individual_score_bbt]
+)
+class TestIndividualScores:
+    def test_all_comparisons_zero(self, compute_method):
+        comparisons = pd.DataFrame(
+            [
+                {
+                    "entity_a": "video_1",
+                    "entity_b": "video_2",
+                    "score": 0,
+                },
+                {
+                    "entity_a": "video_2",
+                    "entity_b": "video_3",
+                    "score": 0,
+                },
+            ]
+        )
+        scores = compute_method(comparisons)
+        for video in ["video_1", "video_2", "video_3"]:
+            assert scores.loc[video].raw_score == 0
+            assert scores.loc[video].raw_uncertainty > 0
+
+    def test_comparisons_chain(self, compute_method):
+        comparisons = pd.DataFrame(
+            [
+                {
+                    "entity_a": f"video_{idx:02d}",
+                    "entity_b": f"video_{idx+1:02d}",
+                    "score": 10,
+                }
+                for idx in range(20)
+            ]
+        )
+        scores = compute_method(comparisons)
+        scores.sort_values("raw_score", inplace=True)
+
+        # Video are ranked according to the chain of comparisons
+        assert list(scores.index) == sorted(scores.index)
+
+        # Compared videos are indexed 0 to 20 included; video_10 is in the middle with score 0.0
+        assert scores.loc["video_10"].raw_score == pytest.approx(0.0, abs=1e-4)
+        for idx in range(10):
+            assert scores.iloc[idx].raw_score == pytest.approx(-1 * scores.iloc[20-idx].raw_score, abs=1e-4)
+
+    def test_individual_scores_mean_is_zero(self, compute_method):
+        comparisons = [
+            ("A", "B", random.randint(-10, 10)),
+            ("A", "C", random.randint(-10, 10)),
+            ("A", "D", random.randint(-10, 10)),
+            ("B", "C", random.randint(-10, 10)),
+            ("B", "D", random.randint(-10, 10)),
+            ("C", "D", random.randint(-10, 10)),
+        ]
+        comparisons_df = pd.DataFrame(comparisons, columns=["entity_a", "entity_b", "score"])
+        scores = compute_method(comparisons_df)
+        assert scores.raw_score.mean() == pytest.approx(1.0, abs=1e-4)
```

Comment:
I don't understand that test. Why do we check that it is "1.0" instead of 0 ? And why do we pick random integers, instead of just the same arbitrary number each time ?

Reply:
`1.0` instead of `0.0` was a typo.  
The random values are used to show that the property is verified for any comparison set.

Reply:
Yes, the random values actually make sense.